modules/fedimint-ln-client/src/lib.rs

```diff
@@ -916,6 +909,43 @@ impl LightningClientModule {
         Ok(())
     }
 
+    /// Continuously update the gateway cache whenever a gateway expires.
+    ///
+    /// Client integrators are expected to call this function in a spawned task.
+    pub async fn update_gateway_cache_continuously(&self) -> ! {
+        let mut first_time = true;
+
+        const ABOUT_TO_EXPIRE: Duration = Duration::from_secs(30);
+        const EMPTY_GATEWAY_SLEEP: Duration = Duration::from_secs(10 * 60);
+        loop {
+            let gateways = self.list_gateways().await;
+            // TODO: filter gateways by vetted
```

Comment:
Won't be possible after this change anymore, feels like we need a function `select_some_gateway(vetted: Vec<GatewayId>) -> _` that returns *some* gateway, prioritizing vetted ones.

Reply:
we might need `filter_gateways: Fn(&GatewayAnnouncement) -> bool` that tells whether the client user cares about that gateway, so we don't update on ignored gateways expiring